editor/src/components/canvas/canvas-strategies/absolute-move-strategy.spec.browser2.tsx

```diff
@@ -0,0 +1,302 @@
+import {
+  getPrintedUiJsCode,
+  makeTestProjectCodeWithSnippet,
+  renderTestEditorWithCode,
+  TestScenePath,
+} from '../ui-jsx.test-utils'
+import { act, fireEvent } from '@testing-library/react'
+import * as EP from '../../../core/shared/element-path'
+import { selectComponents } from '../../editor/actions/action-creators'
+import { CanvasControlsContainerID } from '../controls/new-canvas-controls'
+import CanvasActions from '../canvas-actions'
+import { createInteractionViaMouse } from './interaction-state'
+import { zeroCanvasPoint } from '../../../core/shared/math-utils'
+import { emptyModifiers } from '../../../utils/modifiers'
+
+describe('Absolute Move Strategy', () => {
+  it('moves absolute positioned element', async () => {
+    const renderResult = await renderTestEditorWithCode(
+      makeTestProjectCodeWithSnippet(`
+        <div style={{ width: '100%', height: '100%' }} data-uid='aaa'>
+          <div
+            style={{ backgroundColor: '#0091FFAA', position: 'absolute', left: 40, top: 50, width: 200, height: 120 }}
+            data-uid='bbb'
+            data-testid='bbb'
+          />
+        </div>
+      `),
+      'await-first-dom-report',
+    )
+
+    const targetElement = renderResult.renderedDOM.getByTestId('bbb')
+    const targetElementBounds = targetElement.getBoundingClientRect()
+    const canvasControlsLayer = renderResult.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+    fireEvent(
+      canvasControlsLayer,
+      new MouseEvent('mousedown', {
+        bubbles: true,
+        cancelable: true,
+        metaKey: true,
+        clientX: targetElementBounds.left + 5,
+        clientY: targetElementBounds.top + 5,
+        buttons: 1,
+      }),
+    )
+
+    fireEvent(
+      canvasControlsLayer,
+      new MouseEvent('mousemove', {
+        bubbles: true,
+        cancelable: true,
+        metaKey: false,
+        clientX: targetElementBounds.left + 45,
```

Comment:
because you start the mouse interaction from +5,+5 it's quite  hard to read that the actual "drag" is 40,-25

zooming out, I think these three fireEvents could be grouped into a helper function with a human readable name
both #2339 and #2340 have some examples for helper functions that are aimed at test readability / glanceablility

Reply:
fixed